### PR TITLE
Helpers for ha tests

### PR DESCRIPTION
**Pull request by Mehdi-Bendriss**
## Proposal
This PR adds helpers useful and used in HA tests. Namely, this PR adds:
- `helper_data.py`, helper for data related operations
- Shards collection / operations
- added the `ha-integration` in ci and tox.ini

### CODE REVIEW

**Review by delgod — APPROVED** [review]

File: tests/integration/ha/helpers_data.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helper functions for data related tests, such as indexing, searching etc.."""
```
Comment (MiaAltieri) [inline tests/integration/ha/helpers_data.py:5]: i'm not the most familiar with opensearch calls and such @WRFitch might be able to better review this file

**Review by MiaAltieri — APPROVED** [review]

File: tests/integration/ha/helpers_data.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        doc = default_doc(index_name, doc_id)
+
+    await http_request(
+        ops_test, "PUT", f"https://{unit_ip}:9200/{index_name}/{doc_id}", payload=doc
+    )
+
+    # a refresh makes the indexed data available for search, runs by default every 30 sec,
+    # but we can manually trigger it like below
+    if refresh:
+        await http_request(ops_test, "POST", f"https://{unit_ip}:9200/{index_name}/_refresh")
```
Comment (WRFitch) [inline tests/integration/ha/helpers_data.py:110]: nice

File: tests/integration/ha/helpers_data.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+
+@retry(
+    wait=wait_fixed(wait=5) + wait_random(0, 5),
+    stop=stop_after_attempt(15),
+)
+async def get_doc(ops_test: OpsTest, unit_ip: str, index_name: str, doc_id: int) -> Dict[str, Any]:
+    """Index a simple document."""
```
Comment (WRFitch) [inline tests/integration/ha/helpers_data.py:118]: copy/paste error on this docstring

File: tests/integration/ha/helpers_data.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        ops_test, "GET", f"https://{unit_ip}:9200/{index_name}/_doc/{doc_id}"
+    )
+
+
+@retry(
+    wait=wait_fixed(wait=5) + wait_random(0, 5),
+    stop=stop_after_attempt(15),
+)
+async def delete_doc(ops_test: OpsTest, unit_ip: str, index_name: str, doc_id: int) -> None:
+    """Index a simple document."""
```
Comment (WRFitch) [inline tests/integration/ha/helpers_data.py:129]: copy/paste error on this docstring

File: tests/integration/ha/helpers_data.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        f"https://{unit_ip}:9200/{index_name}/_doc/{doc_id}",
+    )
+
+
+@retry(
+    wait=wait_fixed(wait=5) + wait_random(0, 5),
+    stop=stop_after_attempt(15),
+)
+async def delete_index(ops_test: OpsTest, unit_ip: str, index_name: str) -> None:
+    """Index a simple document."""
```
Comment (WRFitch) [inline tests/integration/ha/helpers_data.py:142]: copy/paste error on this docstring

**Review by WRFitch — APPROVED** [review]

**Pull request merged**